src/pyodide/internal/asgi.py

```diff
@@ -99,53 +99,118 @@ async def send(got):
 
 
 async def process_request(app, req, env):
-    from js import Object, Response
+    from js import Object, Response, TransformStream
 
     from pyodide.ffi import create_proxy
 
     status = None
     headers = None
     result = Future()
-
-    async def response_gen():
-        if req.body:
-            async for data in req.body:
-                yield {
+    is_sse = False
+    finished_response = Event()
+
+    receive_queue = Queue()
+    if req.body:
+        async for data in req.body:
+            await receive_queue.put(
+                {
                     "body": data.to_bytes(),
                     "more_body": True,
                     "type": "http.request",
                 }
-        yield {"body": b"", "more_body": False, "type": "http.request"}
-
-    responses = response_gen()
+            )
+    await receive_queue.put({"body": b"", "more_body": False, "type": "http.request"})
 
     async def receive():
-        return await anext(responses)
+        message = None
+        if not receive_queue.empty():
+            message = await receive_queue.get()
+        else:
+            await finished_response.wait()
+            message = {"type": "http.disconnect"}
+        return message
+
+    # Create a transform stream for handling streaming responses
+    transform_stream = TransformStream.new()
+    readable = transform_stream.readable
+    writable = transform_stream.writable
+    writer = writable.getWriter()
 
     async def send(got):
         nonlocal status
         nonlocal headers
+        nonlocal is_sse
+
         if got["type"] == "http.response.start":
             status = got["status"]
             # Like above, we need to convert byte-pairs into string explicitly.
             headers = [(k.decode(), v.decode()) for k, v in got["headers"]]
-        if got["type"] == "http.response.body":
-            # intentionally leak body to avoid a copy
-            #
-            # Apparently the `Response` constructor will not eagerly copy a
-            # TypedArray argument, so if we don't leak the argument the response
-            # body is corrupted. This is fine because the session is going to
-            # exit at this point.
-            px = create_proxy(got["body"])
+            # Check if this is a server-sent events response
+            for k, v in headers:
+                if k.lower() == "content-type" and v.lower().startswith(
+                    "text/event-stream"
+                ):
+                    is_sse = True
+
+                    # For SSE, create and return the response immediately after http.response.start
+                    resp = Response.new(
+                        readable, headers=Object.fromEntries(headers), status=status
+                    )
+                    result.set_result(resp)
+                    break
```

Comment:
I think the following is clearer:
```suggestion
                    break

            if is_sse:
                # For SSE, create and return the response immediately after http.response.start
                resp = Response.new(
                    readable, headers=Object.fromEntries(headers), status=status
                )
                result.set_result(resp)

```